src/lib.rs

```diff
@@ -0,0 +1,57 @@
+use std::fs;
+use std::io::Result;
+use std::path::{Path, PathBuf};
+
+pub struct Config {
+    path: PathBuf,
+    root: String,
+}
+
+impl Config {
+    pub fn new(root: impl Into<String>, path: impl Into<String>) -> Self {
+        Config {
+            path: Path::new(&path.into()).to_owned(),
+            root: root.into(),
+        }
+    }
+
+    pub fn run(&self, entries: impl IntoIterator<Item = String>) -> Result<()> {
+        for line in entries {
+            let new_path = Path::new(&self.root).join(line);
+            fs::create_dir_all(&new_path)?;
+            println!("Created folder : {}", new_path.display());
+        }
+        Ok(())
+    }
+}
+
+pub mod utils {
+    use std::fs::File;
+    use std::io::{BufRead, BufReader, Result};
+
+    pub fn paths_from_file(file_path: &str) -> Result<impl IntoIterator<Item = String>> {
+        let lines = {
+            let file = File::open(file_path)?;
```

Comment:
This block of code looks interesting.
So, basically, file and reader are going to be temporary variables and the lines variable is going to receive the "collected" lines from the reader?

Also could explain this syntax a bit? Especially the _ 
reader.lines().collect::<Result<Vec<_>>>()?

Reply:
Yes, exactly you got it. My whole idea was that it reads better. Why? Because it *tells* you "hey, file and bufreader are just some means to an ends. Don't get too hooked up on them, their just there so that we can achieve our goal, read the lines, they won't be used later". I know the method is very short, and thus maybe it's not so important, but I still feel it makes it more readable, by telling you what is important and what is there only to achieve something.

Sometimes in C programming you se 1000 variables declared at the **begging**  of a 1000 line function, even known they are only going to be used way later down the road, and then never used again. It makes me confuse.